OneCommunityGlobal/HighestGoodNetworkApp
# Rahul - Added Frontend Files For The HGN Form

# Description
![Screenshot 2024-12-05 at 3 45 29 PM](https://github.com/user-attachments/assets/0350e21a-3388-4b49-8430-48b1d9b866fe)

This PR contains the frontend code for the HGN Form application, designed to help developers assess their skill sets across various parameters, promoting collaboration by enabling them to offer or seek assistance from others. The PR focuses on the React.js-based frontend of the form application, which includes six pages. The implementation is demonstrated in the video below. This form mimics the Google Form [HGN Development Team Questionnaire](https://forms.gle/GCnqZdNvTpMbs4TT9). The Figma designs for this application can be found [here](https://www.figma.com/design/TweXhwOF7tipvogbiZqMLi/OneCommunityGlobal?node-id=0-1).

## Related PR
This frontend PR is related to the backend PR [#1164](https://github.com/OneCommunityGlobal/HGNRest/pull/1164).
Before starting this application, make sure to run the backend. Follow the [link](https://github.com/OneCommunityGlobal/HGNRest/pull/1164) for instructions on how to start the backend.

## Main features explained
- Developed Frontend Form Pages 1 to 6 as per the Figma designs.
- Implemented functionality for the Owner to edit the form questions.
- Enabled Users to edit previously answered questions by using the "Back" or "Edit Response" buttons on Page 5.
- Ensured the form is responsive, incorporating auto-scrolling and smooth page transitions.
- Added validation and checks for the respective form fields.
- Implemented a progress bar to track the user's form completion.
- Utilized React Context to preserve form state across pages, allowing users to edit their responses.
- Implemented smart selection for Availability and Preferences fields, including grouping consecutive times.
- Organized the form data collected across pages before sending it to the backend.

## How to test
Before starting this application, make sure to run the backend. Follow the [link](https://github.com/OneCommunityGlobal/HGNRest/pull/1164) for instructions on how to start the backend. You can also check out this [video](https://www.loom.com/share/e8a28ca8345a4938b8348a208311adb1?sid=e8405e40-d4aa-4c2a-9c05-a1c4b8af2d0c) for starting the backend. Currently, the Owner mode is set to true in the code to allow everyone to test the "edit questions" feature.
1. Switch to the current branch using: `git checkout Rahul-HGN-Form`
2. Navigate to the directory with: `cd src/components/HGNForm/frontend`
3. Install the required dependencies by running: `npm install`
4. Start the application by running: `npm start`
5. Complete the form and check the output in your browser's console.
6. Verify that the features mentioned above are functioning as expected.
7. Watch the video below to see how the form works.

## Video Demo of the HGN Form
https://www.loom.com/share/47aafbd487724dad8f6000a9f6342982?sid=a2151be0-5708-4d86-b35a-25b6da6bf3b8
